### PR TITLE
go 1.17 and rocksdb

### PR DESCRIPTION
**Pull request by faddat**
* requires go 1.17
* begins work to make rocksdb testable by using a replace for cosmos/gorocksdb
* updates readme.md to reflect requirement of go 1.17
* adds benchmarks to github actions ci
* update .golang-ci.yml
* allow the lint action to use the latest version of .golang-ci.yml

### CODE REVIEW

**Review by marbar3778 — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -12,8 +12,32 @@ require (
...
+	github.com/tecbot/gorocksdb v0.0.0-20191217155057-f0fad39f321c // indirect
+	go.etcd.io/bbolt v1.3.6 // indirect
+	golang.org/x/net v0.0.0-20211005001312-d4b1ae081e3b // indirect
+	golang.org/x/sys v0.0.0-20211013075003-97ac67df715c // indirect
+	golang.org/x/text v0.3.7 // indirect
+	google.golang.org/protobuf v1.27.1 // indirect
+	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
+)
+
+replace github.com/tecbot/gorocksdb => github.com/cosmos/gorocksdb v1.2.0
```
Comment (marbar3778) [inline go.mod:43]: ah whoops this isn't needed actually. tm-db is used for testing here and the interface. it has nothing to do with gorocksdb.

Comment (faddat) [reply go.mod:43]: ah, ok-- I'll change the imports to reflect the new repo

Comment (faddat) [reply go.mod:43]: fairly sure we will need it for benchmarks now that I've checked

Comment (robert-zaremba) [reply go.mod:43]: let's use the same one as we do in the SDK, so I'm for keeping this replace

**Review by robert-zaremba — APPROVED** [review]
lgtm

**Review by robert-zaremba — APPROVED** [review]
thanks!

**Comment by robert-zaremba** [comment]
There are few linter issues to fix

**Comment by robert-zaremba** [comment]
closing this PR - some things were already implemented in other PRs

**Pull request closed without merging**